domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/Attribution.java

```diff
@@ -0,0 +1,168 @@
+package org.mobilitydata.gtfsvalidator.domain.entity;
+
+import org.jetbrains.annotations.NotNull;
+
+public class Attribution {
+
+    private final String attributionId;
+    private final String agencyId;
+    private final String routeId;
+    private final String tripId;
+
+    @NotNull
+    private final String organizationName;
+
+    private final IsProducer isProducer;
+    private final IsOperator isOperator;
+    private final IsAuthority isAuthority;
+    private final String attributionUrl;
+    private final String attributionEmail;
+    private final String attributionPhone;
+
+    public Attribution(String attributionId,
+                       String agencyId,
+                       String routeId,
+                       String tripId,
+                       @NotNull String organizationName,
+                       IsProducer isProducer,
+                       IsOperator isOperator,
+                       IsAuthority isAuthority,
+                       String attributionUrl,
+                       String attributionEmail,
+                       String attributionPhone) {
+        this.attributionId = attributionId;
+        this.agencyId = agencyId;
+        this.routeId = routeId;
+        this.tripId = tripId;
+        this.organizationName = organizationName;
+        this.isProducer = isProducer;
+        this.isOperator = isOperator;
+        this.isAuthority = isAuthority;
+        this.attributionUrl = attributionUrl;
+        this.attributionEmail = attributionEmail;
+        this.attributionPhone = attributionPhone;
+    }
+
+    public String getAttributionId() {
+        return attributionId;
+    }
+
+    public String getAgencyId() {
+        return agencyId;
+    }
+
+    public String getRouteId() {
+        return routeId;
+    }
+
+    public String getTripId() {
+        return tripId;
+    }
+
+    @NotNull
+    public String getOrganizationName() {
+        return organizationName;
+    }
+
+    public IsProducer getIsProducer() {
+        return isProducer;
+    }
+
+    public IsOperator getIsOperator() {
+        return isOperator;
+    }
+
+    public IsAuthority getIsAuthority() {
+        return isAuthority;
+    }
+
+    public String getAttributionUrl() {
+        return attributionUrl;
+    }
+
+    public String getAttributionEmail() {
+        return attributionEmail;
+    }
+
+    public String getAttributionPhone() {
+        return attributionPhone;
+    }
+
+    public static class AttributionBuilder {
+        private String attributionId;
+        private String agencyId;
+        private String routeId;
+        private String tripId;
+        private String organizationName;
+        private IsProducer isProducer;
+        private IsOperator isOperator;
+        private IsAuthority isAuthority;
```

Comment:
One of those three fields is mandatory set to 1. This fact should be translated as to have different *classes*, on the model of what has been done for stops entities.

Reply:
A nuance being that "at least one"of these fields must be set to 1. 

Let's imagine that a row from attributions.txt has both isOperator and isProducer booleans set to 1: following the model of what's been done for stop entities could lead to loss of information in a way that that either we would have a ProducerAttribution or an OperatorAttribution entity.

In the use case "ProcessParsedAttribution", shall we consider using a vector of three booleans (isProducer, isOperator, isAuthority) named as "attributionRole"? From that vector we could set fields isProducer, isOperator, and isAuthority (via a builder pattern) depending of the components of said vector. 

The case (0,0,0): all fields set to 0,  would generate a notice in the (future) use case ProcessParsedAttribution. 

This would allow us to only have one class "Attribution" instead of 7 sub-classes depending on a parent class "AttributionBase".